lib/puppet/provider/cisco_stp_global/cisco.rb

```diff
@@ -0,0 +1,298 @@
+  PuppetX::Cisco::AutoGen.mk_puppet_methods(:bool, self, '@nu',
+                                            STP_GLOBAL_BOOL_PROPS)
+  PuppetX::Cisco::AutoGen.mk_puppet_methods(:array_flat, self, '@nu',
+                                            STP_GLOBAL_ARRAY_FLAT_PROPS)
+
+  def initialize(value={})
+    super(value)
+    @nu = Cisco::StpGlobal.globals[@property_hash[:name]]
+    @property_flush = {}
+    debug 'Created provider instance of cisco_stp_global'
+  end
+
+  def self.properties_get(global_id, nu_obj)
+    debug "Checking instance, global #{global_id}"
+    current_state = {
+      name: global_id
+    }
+
+    # Call node_utils getter for each property
+    STP_GLOBAL_NON_BOOL_PROPS.each do |prop|
+      current_state[prop] = nu_obj.send(prop)
+    end
+    STP_GLOBAL_BOOL_PROPS.each do |prop|
+      val = nu_obj.send(prop)
+      if val.nil?
+        current_state[prop] = nil
+      else
+        current_state[prop] = val ? :true : :false
+      end
+    end
+    STP_GLOBAL_ARRAY_FLAT_PROPS.each do |prop|
+      current_state[prop] = nu_obj.send(prop)
+    end
+    # nested array properties
+    current_state[:bd_designated_priority] = nu_obj.bd_designated_priority
+    current_state[:bd_forward_time] = nu_obj.bd_forward_time
+    current_state[:bd_hello_time] = nu_obj.bd_hello_time
+    current_state[:bd_max_age] = nu_obj.bd_max_age
+    current_state[:bd_priority] = nu_obj.bd_priority
+    current_state[:bd_root_priority] = nu_obj.bd_root_priority
+    current_state[:mst_designated_priority] = nu_obj.mst_designated_priority
+    current_state[:mst_inst_vlan_map] = nu_obj.mst_inst_vlan_map
+    current_state[:mst_priority] = nu_obj.mst_priority
+    current_state[:mst_root_priority] = nu_obj.mst_root_priority
+    current_state[:vlan_designated_priority] = nu_obj.vlan_designated_priority
+    current_state[:vlan_forward_time] = nu_obj.vlan_forward_time
+    current_state[:vlan_hello_time] = nu_obj.vlan_hello_time
+    current_state[:vlan_max_age] = nu_obj.vlan_max_age
+    current_state[:vlan_priority] = nu_obj.vlan_priority
+    current_state[:vlan_root_priority] = nu_obj.vlan_root_priority
+    new(current_state)
+  end # self.properties_get
+
+  def self.instances
+    globals = []
+    Cisco::StpGlobal.globals.each do |global_id, nu_obj|
+      globals << properties_get(global_id, nu_obj)
+    end
+    globals
+  end
+
+  def self.prefetch(resources)
+    globals = instances
+
+    resources.keys.each do |id|
+      provider = globals.find { |nu_obj| nu_obj.instance_name == id }
+      resources[id].provider = provider unless provider.nil?
+    end
+  end # self.prefetch
+
+  def instance_name
+    name
+  end
+
+  def properties_set
+    STP_GLOBAL_ALL_PROPS.each do |prop|
+      next unless @resource[prop]
+      unless @property_flush[prop].nil?
+        @nu.send("#{prop}=", @property_flush[prop]) if
+          @nu.respond_to?("#{prop}=")
+      end
+    end
+  end
+
+  def bd_designated_priority=(should_list)
```

Comment:
:+1: Looks much better with the autogened getters!  Will look even better in the future when we can autogen these setters :smile: !